lib/bundler/uri_credentials_filter.rb

```diff
@@ -5,7 +5,7 @@ module URICredentialsFilter
 
     def credential_filtered_uri(uri_to_anonymize)
       return uri_to_anonymize if uri_to_anonymize.nil?
-      uri = uri_to_anonymize.dup
+      uri = uri_to_anonymize.clone
```

Comment:
I'm generally against the principle of altering application code purely for the sake of testing. What necessitates this change from `dup` to `clone`? Is it possible to replace the mocks in whichever tests this is affecting with actual URI objects?